moc_openshift.py

```diff
@@ -469,3 +541,108 @@ def update_rolebindings(self, project_name, role, rolebindings_json):
         return self.put_request(url, payload, True)
+
+    def get_moc_quota(self, project_name):
+        quota_def = self.get_quota_definitions()
+        quota = {}
+        for quota_name in quota_def:
+            quota[quota_name] = quota_def[quota_name]["value"]
+
+        quota_object = {
+            "Version": "0.9",
+            "Kind": "MocQuota",
+            "ProjectName": project_name,
+            "Quota": quota,
+        }
+        return quota_object
+
+    def create_shift_quotas(self, project_name, quota_spec):
+        quota_def = {}
+        # separate the quota_spec by quota_scope
+        for mangled_quota_name in quota_spec:
+            (scope, quota_name) = self.split_quota_name(mangled_quota_name)
+            if scope not in quota_def:
+                quota_def[scope] = {}
+            quota_def[scope][quota_name] = quota_spec[mangled_quota_name]
+        # create the openshift quota resources
+        quota_create_status_code = 200
+        quota_create_msg = ""
+        for scope, quota_item in quota_def.items():
+            resource_quota_json = {
+                "apiVersion": "v1",
+                "kind": "ResourceQuota",
+                "metadata": {"name": f"{project_name.lower()}-{scope.lower()}"},
+                "spec": {"hard": {}},
+            }
+            if scope != "Project":
+                resource_quota_json["spec"]["scopes"] = []
+                resource_quota_json["spec"]["scopes"].append(scope)
+            non_null_quota_count = 0
+            for quota_name in quota_item:
+                if quota_item[quota_name] is not None:
+                    non_null_quota_count += 1
+                    resource_quota_json["spec"]["hard"][quota_name] = quota_item[
+                        quota_name
+                    ]["value"]
+            if non_null_quota_count > 0:
+                url = (
+                    f"{self.get_url()}/api/v1/namespaces/{project_name}/resourcequotas"
+                )
+                resp = self.post_request(url, resource_quota_json, True)
+                time.sleep(2)
+                if resp.status_code in [200, 201]:
+                    quota_create_msg = f"{quota_create_msg} Quota {project_name}/{quota_name} successfully created\n"
+                else:
+                    if resp.status_code > quota_create_status_code:
+                        quota_create_status_code = resp.status_code
+                quota_create_msg = f"{quota_create_msg} Quota {project_name}/{quota_name} creation failed"
+            if quota_create_status_code == 200:
+                quota_create_msg = f"All quota from {project_name} successfully created"
+        return Response(
+            response=quota_create_msg,
+            status=quota_create_status_code,
+            mimetype="application/json",
+        )
+
+    def get_resourcequotas(self, project_name) -> list:
+        """Returns a dictionary of all of the resourcequota objects"""
```

Comment:
looks like you are returning a list, and not a dictionary here.

Reply:
And this is why I like the return hints - unfortunately they are just hints that python ignores.